src/block/mod.rs

```diff
@@ -182,24 +182,19 @@ pub fn parse_file(filename: &Path) -> anyhow::Result<PrecomputedBlock> {
     } else {
         Err(anyhow!(
             "Unable to parse invalid precomputed block: {}",
-            filename.display()
+            path.display()
         ))
     }
 }
 
 /// Extracts a state hash from an OS file name
 pub fn get_state_hash(file_name: &OsStr) -> Option<String> {
     let last_part = file_name.to_str()?.split('-').last()?.to_string();
-    if last_part.starts_with('.') || !last_part.starts_with("3N") {
-        return None;
-    }
-
     let state_hash = last_part.split('.').next()?;
-    if state_hash.contains('.') {
-        return None;
+    if state_hash.starts_with("3N") {
```

Comment:
Significance of "3N" ?

Reply:
Just some bytes to distinguish state hashes from the other various types of hashes used

Reply:
Might it be helpful to capture the type of hash in a variable?